### PR TITLE
Use interfaces for input parameters type

### PR DESCRIPTION
**Pull request by popenta**

### CODE REVIEW

File: multiversx_sdk_core/tokens.py [hunk 1]
```diff
@@ -5,6 +5,7 @@
 from multiversx_sdk_core.constants import TOKEN_RANDOM_SEQUENCE_LENGTH
 from multiversx_sdk_core.errors import (BadUsageError,
                                         InvalidTokenIdentifierError)
+from multiversx_sdk_core.interfaces import IToken, ITokenIdentifierParts
 
 
 class Token:
```
Comment (ssd04) [inline multiversx_sdk_core/tokens.py:11]: use `IToken` in `TokenTransfer` constructor?

Comment (popenta) [reply multiversx_sdk_core/tokens.py:11]: indeed. fixed

File: multiversx_sdk_core/interfaces.py [hunk 2]
```diff
@@ -8,6 +8,48 @@ def to_bech32(self) -> str: ...
...
+    signature: bytes
+
+
+class IToken(Protocol):
+    identifier: str
+    nonce: int
+
+
+class ITokenTransfer(Protocol):
+    @property
```
Comment (andreibancioiu) [inline multiversx_sdk_core/interfaces.py:40]: So only this was needed to be defined as a read-only property?

**Review by andreibancioiu — APPROVED** [review]

**Review by ssd04 — APPROVED** [review]

File: multiversx_sdk_core/interfaces.py (hunk 2 above)
Comment (popenta) [reply multiversx_sdk_core/interfaces.py:40]: Yes

**Pull request merged**